frontend/catalyst/jit.py

```diff
@@ -663,6 +664,15 @@ def fn_with_transform_named_sequence(*args, **kwargs):
             jaxpr. It is never executed or used anywhere, except being traced here.
             """
             _inject_transform_named_sequence()
+
+            if self.user_function.device == "ionic":
```

Comment:
Two concerns with adding the pass to the transform sequence here in preprocessing of tracing:

1. Is the preprocessing function getting too complicated? For example in the future when we have multiple device decompositions, there would be a bunch of if else blocks here, making the preprocessing of the traced target much less readable. Maybe we can have a new function, `get_device_decomposition_pass()`, and perform this logic somewhere else, and just call that logic here?

2. @josh146 should the ion gateset decomposition pass expose a direct frontend API (like cancel inverse and merge rotations)? In that case we just call the pass decorator here.

Reply:
I agree this doesn't seem like the right place for it. This module provides the general structure and high-level process for performing jit compilation with catalyst (using the [template pattern](https://en.wikipedia.org/wiki/Template_method_pattern) to allow for customization).
We certainly should not be calling `bind` functions here which belong to the tracing part of the stack.

Reply:
I talked about that with @dime10 and the current solution is to just match the device name. In the future, we need a good way to add that from the TOML file.